python/nav/web/alertprofiles/views.py

```diff
@@ -1575,15 +1577,32 @@ def filter_saveexpression(request):
             request, _('You do not own this filter.')
         )
 
+    if match_field.data_type == MatchField.IP:
+        if operator.type == Operator.IN:
+            value_list = request.POST.get('value').split()
+        else:
+            value_list = [request.POST.get('value')]
+        for value in value_list:
+            try:
+                IP(value)
+            except ValueError:
+                new_message(
+                    request,
+                    f"Invalid IP address: {value}",
+                    Messages.ERROR,
+                )
+                request.POST = QueryDict(
+                    f"id={request.POST.get('filter')}&matchfield={request.POST.get('match_field')}"
```

Comment:
Directly trusting something with no validation? Ick. Have you considered using a django form to validate?

Reply:
Filter is not user input and `match_field` is chosen from a dropdown list and if `filter` or `match_field` would be invalid we would already get an error in lines 1570-1572.

Reply:
Drop down lists can be changed via browser devtools, to pick the easiest way.

NAV is older than Django. The Django way is to have a form for validation, feed `request.POST` to it, run `form.is_valid()` (which might raise ValidationError) and read the cleaned values off of `form.validated_data`.

`request.POST.get*` is a code-smell. A quick grep says there are 190 instances of it throughout the code base. I personally would like to have a long-running task to form-ify as many of these as possible.

Building up a string to feed to QueryDict is also a code-smell (15 of those found with a braindead grep).

One solution is going via a MultiDict: (https://stackoverflow.com/a/31166557)

```
from django.http.request import QueryDict, MultiValueDict

dictionary = {'my_age': ['23'], 'my_girlfriend_age': ['25', '27'], }
qdict = QueryDict('', mutable=True)
qdict.update(MultiValueDict(dictionary))
```
Another is explicitly url-encoding the string: (https://stackoverflow.com/a/13363676)

```
from urllib.parse import urlencode
from django.http.request import QueryDict

params = {'a': 'one', 'b': 'two', }
urlparams = urlencode(params)
qdict = QueryDict(urlparams)
```

We can do it as is to get it out of the way, but I'm pretty sure a security audit would come to the exact same conclusion as me.